packages/utils/object.ts

```
@@ -16,6 +16,8 @@ export function checkObjectHasProperties(object: any, properties: string[]) {
 	}
 }
 
-export const hasOwnProperty = (object: object, property: string): boolean => {
+export const hasOwnProperty = <T extends object, Property extends string>(
```

Biome (2.3.14)

[error] Do not shadow the global "hasOwnProperty" property.

Consider renaming this variable. It's easy to confuse the origin of variables when they're named after a known global.

(lint/suspicious/noShadowRestrictedNames)